pkg/issuer/acme/http/http.go

```diff
@@ -290,13 +294,13 @@ func testReachability(ctx context.Context, url *url.URL, key string, dnsServers
 		log.V(logf.DebugLevel).Info("failed to perform self check GET request", "error", err)
 		return fmt.Errorf("failed to perform self check GET request '%s': %v", url, err)
 	}
+	defer response.Body.Close()
```

Comment:
Not sure if it is possible that the status code is not 200, yet there is an open connection, but this seems more readable and harmless in either case 👍🏼